framework/src/mesh/MooseMesh.C

```diff
@@ -523,7 +523,12 @@ MooseMesh::updateActiveSemiLocalNodeRange(std::set<dof_id_type> & ghosted_elems)
     const Elem * elem = *it;
     for (unsigned int n = 0; n < elem->n_nodes(); ++n)
     {
-      Node * node = elem->node_ptr(n);
+      // Since elem is const here but we require a non-const Node * to
+      // store in the _semilocal_node_list (otherwise things like
+      // UpdateDisplacedMeshThread don't work), we are using the "old"
+      // Elem interface to get a non-constant Node pointer from a
+      // constant Elem.
+      Node * node = elem->get_node(n);
```

Comment:
I'm not 100% sure, but updateActiveSemiLocalNodeRange() doesn't seem like it needs to iterate over a ConstElemRange necessarily to build up `_semilocal_node_list`.  So my plan was to change that to non-const and see how it goes...